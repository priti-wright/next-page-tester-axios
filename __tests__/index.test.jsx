
import React from 'react'
import { render, screen } from '@testing-library/react'
import Home from '../pages/index'
import { getPage } from "next-page-tester";
import MockAdapter from "axios-mock-adapter";

const axios = require("axios");
const mock = new MockAdapter(axios);

describe('Home', () => {
  beforeEach(() => {
    mock.onGet(`${process.env.NEXT_PUBLIC_URL}api/name`).reply(200, {name: "Tester"});
  });
  it('renders a heading (passes)', () => {
    render(<Home />)

    const heading = screen.getByRole('heading', {
      name: /welcome to next\.js/i,
    });

    expect(heading).toBeInTheDocument();
  })

  it("renders correctly with next-page-tester", async () => {
    const { page } = await getPage({
      route: `/`,
    });

    render(page);

    const heading = screen.getByRole('heading', {
      name: /welcome to next\.js/i,
    })
    expect(heading).toBeInTheDocument();
  });

  it("renders correctly with api call", async () => {
    const { page } = await getPage({
      route: `/`,
    });

    render(page);

    const heading = screen.getByRole('heading', {
      name: /welcome to next\.js Tester/i,
    });

  });

  it("renders a snapshot correctly", async () => {
    const { page } = await getPage({
      route: `/`,
    });
    const { container } = render(page);

    expect(
      await screen.findByRole("heading", { 
        name: /welcome to next\.js Tester/i
      })
    ).toBeVisible();

    expect(container).toMatchSnapshot();
  });
})
