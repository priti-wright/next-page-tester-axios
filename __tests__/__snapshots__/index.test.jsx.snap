// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders a snapshot correctly 1`] = `
<div>
  <div
    class="container"
  >
    <main>
      <h1
        class="title"
      >
        Welcome to 
        <a
          href="https://nextjs.org"
        >
          Next.js
        </a>
         
        Tester
        !
      </h1>
      <p
        class="description"
      >
        Get started by editing 
        <code>
          pages/index.js
        </code>
      </p>
    </main>
  </div>
</div>
`;
